packages/gatsby-source-graphql/README.md

```diff
@@ -176,5 +176,162 @@ module.exports = {
+
+**Caveat**: Batching is only possible for queries starting approximately at the same time. In other words
+it is bounded by the number of parallel GraphQL queries executed by Gatsby (by default it is **4**).
+
+Fortunately we can increase the number of queries executed in parallel by setting [environment variable](https://gatsby.dev/env-vars)
+`GATSBY_EXPERIMENTAL_QUERY_CONCURRENCY` to a higher value and setting `batch` option of the plugin
+to `true`.
+
+Example:
+
+```
+cross-env GATSBY_EXPERIMENTAL_QUERY_CONCURRENCY=20 gatsby develop
+```
+
+With plugin config:
+
+```js
+const fs = require("fs")
+const { buildSchema, buildClientSchema } = require("graphql")
+
+module.exports = {
+  plugins: [
+    {
+      resolve: "gatsby-source-graphql",
+      options: {
+        typeName: "SWAPI",
+        fieldName: "swapi",
+        url: "https://api.graphcms.com/simple/v1/swapi",
+        batch: true,
+      },
+    },
+  ],
+}
+```
+
+By default, the plugin batches up to 5 queries. You can override this by passing
+`dataLoaderOptions`:
+
+```js
+const fs = require("fs")
+const { buildSchema, buildClientSchema } = require("graphql")
+
+module.exports = {
+  plugins: [
+    {
+      resolve: "gatsby-source-graphql",
+      options: {
+        typeName: "SWAPI",
+        fieldName: "swapi",
+        url: "https://api.graphcms.com/simple/v1/swapi",
+        batch: true,
+        // See https://github.com/graphql/dataloader#new-dataloaderbatchloadfn--options
+        // for a full list of DataLoader options
+        dataLoaderOptions: {
+          maxBatchSize: 10,
+        },
+      },
+    },
+  ],
+}
+```
+
+Having 20 parallel queries with 5 queries per batch means we are still running 4 batches
+in parallel.
+
+Each project is unique so try tuning those two variables and see what works best for you.
+We've seen up to 5-10x speed-up for some setups.
+
+### How batching works
+
+Under the hood `gatsby-source-graphql` uses [DataLoader](https://github.com/graphql/dataloader)
+for query batching. It merges all queries from a batch to a single query that gets sent to the
+server in a single network request.
+
+Consider following example:
+
+```js
+{
+  query: `query(id: Int!) {
+    node(id: $id) {
+      foo
+    }
+  }`,
+  variables: { id: 1 },
+}
+```
+
+```js
+{
+  query: `query(id: Int!) {
+    node(id: $id) {
+      bar
+    }
+  }`,
+  variables: { id: 2 },
+}
+```
+
+They will be merged to a single query:
+
+```js
+{
+  query: `
+    query(
+      $gatsby0_id: Int!
+      $gatsby1_id: Int!
+    ) {
+      gatsby0_node: node(id: $gatsby0_id) {
+        foo
+      }
+      gatsby1_node: node(id: $gatsby1_id) {
+        bar
+      }
+    }
+  `,
+  variables: {
+    gatsby0_id: 1,
+    gatsby1_id: 2,
+  }
+}
+```
+
+Then `gatsby-source-graphql` splits the result of this single query to multiple results
+and delivers it back to Gatsby as if it executed multiple queries:
+
+```js
```

Comment:
```suggestion
```json
```

Reply:
At the moment this example is a valid JS object but not a valid JSON. I guess I can change it to JSON but I thought it is more readable this way?

```js
{
  data: {
    gatsby0_node: { foo: `foo` },
    gatsby1_node: { bar: `bar` },
  },
}
```
vs
```json
{
  "data": {
    "gatsby0_node": { "foo": "foo" },
    "gatsby1_node": { "bar": "bar" }
  }
}
```
What do you think?